src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -304,7 +304,8 @@ Handle java_lang_String::create_from_unicode(const jchar* unicode, int length, T
 #ifdef ASSERT
   {
     ResourceMark rm;
-    char* expected = UNICODE::as_utf8(unicode, length);
+    size_t utf8_len = static_cast<size_t>(length);
```

Comment:
I think there should be an assert that length is not negative, probably at the very beginning of this function.

Reply:
Why? As I explained to Coleen this is the length obtained from a Java array. All of the existing code relies on length being >= 0 and doesn't assert that anywhere.

Reply:
If called from jni_NewString --> java_lang_String::create_oop_from_unicode, there's no Java String yet.  Yes, it has probably been like this forever.